helix-core/src/main/java/org/apache/helix/controller/GenericHelixController.java

```diff
@@ -703,9 +705,15 @@ private void handleEvent(ClusterEvent event, BaseControllerDataProvider dataProv
     event.addAttribute(AttributeName.STATEFUL_REBALANCER.name(),
         _rebalancerRef.getRebalancer(manager));
 
-    if (!manager.isLeader()) {
-      logger.error("Cluster manager: " + manager.getInstanceName() + " is not leader for " + manager
-          .getClusterName() + ". Pipeline will not be invoked");
+    // If manager session changes, no need to run pipeline for the stale event.
```

Comment:
This is right, but the original logic was, if the manager is not leader, do not run the pipeline. Where is that logic going?